### PR TITLE
Make size_average docs clearer

### LINKED ISSUE #6988: The loss computation with `size_average` should average over batch example or batch element ?
Traditionally, when we have a batched data with shape `(N, D)` where `N` is batch size and `D` is data dimension. The losses are often calculated for each training example say
```
L_i = loss(X_i), i = 1, ..., N
```
And then total loss is averaged over the batch size
```
L = (1/N)*sum(L_i)
```

However, it seems it is not what `nn.*Loss` is doing for the flag `size_average=True/False`

e.g. 

```python
import torch
import torch.nn.functional as F

input = torch.randn(3, 2)
target = torch.rand(3, 2)

print(input)
print(target)

full_loss = F.mse_loss(input, target, reduce=False)
print(full_loss)
loss_sum = F.mse_loss(input, target, size_average=False)
loss_mean = F.mse_loss(input, target, size_average=True)
print(loss_sum)
print(loss_mean)

batch_loss = full_loss.sum(dim=-1)
correct_loss_mean = batch_loss.mean()
correct_loss_sum = batch_loss.sum()
print(correct_loss_mean)
print(correct_loss_sum)

assert correct_loss_sum == loss_sum
assert correct_loss_mean == loss_mean
```
it seems the `size_average` does not properly average the loss over the batch of examples, but averaged over all dimensions.

### PR DESCRIPTION
**Pull request by li-roy**
Fix #6988
cc @zou3519

### CODE REVIEW

**Comment by SsnL** [comment]
1. nit: `` :attr:`size_average` ``.
2. IIRC, for some losses, `size_average` does different things than other losses. Does the new doc accurately describe their behaviors?

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -134,9 +135,10 @@ class NLLLoss(_WeightedLoss):
            class. If given, it has to be a Tensor of size `C`. Otherwise, it is
            treated as if having all ones.
         size_average (bool, optional): By default, the losses are averaged
-           over observations for each minibatch with weights set by
-           :attr:`weight`. However, if the field :attr:`size_average` is set to
+           over each loss element in the batch. Note that for some losses, there
```
Comment (zou3519) [inline torch/nn/modules/loss.py]: It might be better to have two versions of the description to clarify if the loss does an average over all elements, or if the loss averages over batch_size.  Users might not understand "losses are averaged over each loss element" at first glance

**Review by soumith — APPROVED** [review]
